src/Razor/Razor.Runtime/src/RazorRazorPageBase.cs

```diff
@@ -0,0 +1,505 @@
+        /// </summary>
+        public virtual TextWriter Output
+        {
+            get
+            {
+                var viewContext = ViewContext;
+                if (viewContext == null)
+                {
+                    throw new InvalidOperationException("View context must be set"); //TODO Resources.FormatViewContextMustBeSet(nameof(ViewContext), nameof(Output)));
+                }
+
+                return viewContext.Writer;
+            }
+        }
+
+        /// <inheritdoc />
+        public string Path { get; set; }
+
+        /// <inheritdoc />
+        public IDictionary<string, RenderAsyncDelegate> SectionWriters { get; } =
+            new Dictionary<string, RenderAsyncDelegate>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Gets the dynamic view data dictionary.
+        /// </summary>
+        public dynamic ViewBag => ViewContext?.ViewBag;
+
+        /// <inheritdoc />
+        public bool IsLayoutBeingRendered { get; set; }
+
+        /// <inheritdoc />
+        public IHtmlContent BodyContent { get; set; }
+
+        /// <inheritdoc />
+        public IDictionary<string, RenderAsyncDelegate> PreviousSectionWriters { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="System.Text.Encodings.Web.HtmlEncoder"/> to use when this <see cref="RazorRazorPageBase{TOutputContext}"/>
+        /// handles non-<see cref="IHtmlContent"/> C# expressions.
+        /// </summary>
+        public virtual HtmlEncoder HtmlEncoder { get; set; }
+
+        public abstract Task ExecuteAsync();
+
+        // Internal for unit testing.
+        protected internal virtual void PushWriter(TextWriter writer)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            var viewContext = ViewContext;
+            _textWriterStack.Push(viewContext.Writer);
+            viewContext.Writer = writer;
+        }
+
+        // Internal for unit testing.
+        protected internal virtual TextWriter PopWriter()
+        {
+            var viewContext = ViewContext;
+            var writer = _textWriterStack.Pop();
+            viewContext.Writer = writer;
+            return writer;
+        }
+
+        /// <inheritdoc />
+        public abstract string Href(string contentPath);
+
+        /// <summary>
+        /// Creates a named content section in the page that can be invoked in a Layout page using
+        /// <c>RenderSection</c> or <c>RenderSectionAsync</c>
+        /// </summary>
+        /// <param name="name">The name of the section to create.</param>
+        /// <param name="section">The delegate to execute when rendering the section.</param>
+        /// <remarks>This is a temporary placeholder method to support ASP.NET Core 2.0.0 editor code generation.</remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected void DefineSection(string name, Func<object, Task> section)
```

Comment:
This is an old back-compat placeholder, code generation doesn't use it so it can be removed.